tests/Resources/Page/ODH/ODHDashboard/ODHDashboard.robot

```diff
@@ -86,3 +98,154 @@ Verify Service Provides "Get Started" Button In The Explore Page
 
+Check HTTP Status Code
+    [Arguments]  ${link_to_check}  ${expected}=200
+    ${response}=    RequestsLibrary.GET  ${link_to_check}   expected_status=any
+    Run Keyword And Continue On Failure  Status Should Be  ${expected}
+    [Return]  ${response.status_code}
+
+Load App Info JSON File
+    ${APPS_DICT}=  Load Json File  ${APPS_DICT_PATH}
+    ${APPS_DICT}=  Set Variable  ${APPS_DICT}[apps]
+    Set Suite Variable  ${APPS_DICT}
+
+Wait Until Cards Are Loaded
+    Wait Until Page Contains Element    xpath://div[contains(@class,'odh-explore-apps__gallery')]
+
+Get App ID From Card
+    [Arguments]  ${card_locator}
+    ${id}=  Get Element Attribute    xpath:${card_locator}    id
+    [Return]  ${id}
+
+Check Number of Cards
+    ${N_TILES}=  Get Element Count    xpath:${TILES_XP}
+    Set Suite Variable  ${N_TILES}
+    ${expected_n}=  Get Length    ${APPS_DICT}
+    Run Keyword And Continue On Failure    Should Be Equal  ${N_TILES}  ${expected_n}
+
+Get Card Texts
+    [Arguments]  ${card_locator}
+    ${title}=  Get Text    xpath:${card_locator}/${TITLE_XP}
+    ${provider}=  Get Text    xpath:${card_locator}/${PROVIDER_XP}
+    ${desc}=  Get Text    xpath:${card_locator}/${DESCR_XP}
+    [Return]  ${title}  ${provider}  ${desc}
+
+Check Card Texts
+    [Arguments]  ${card_locator}  ${app_id}
+    ${card_title}  ${card_provider}  ${card_desc}=  Get Card Texts  card_locator=${card_locator}
+    Run Keyword And Continue On Failure  Should Be Equal   ${card_title}  ${APPS_DICT}[${app_id}][title]
+    Run Keyword And Continue On Failure  Should Be Equal   ${card_provider}  ${APPS_DICT}[${app_id}][provider]
+    Run Keyword And Continue On Failure  Should Be Equal   ${card_desc}  ${APPS_DICT}[${app_id}][description]
+
+Get Card Badges Titles
+    [Arguments]  ${card_locator}
+    ${badges}=  Get WebElements    xpath:${card_locator}/${BADGES_XP}
+    ${badges_titles}=  Create List
+    FOR    ${cb}    IN    @{badges}
+        ${btitle}=  Get Text   ${cb}
+        Append To List    ${badges_titles}  ${btitle}
+    END
+    [Return]  ${badges_titles}
+
+Check Card Badges And Return Titles
+    [Arguments]  ${card_locator}  ${app_id}
+    ${card_badges_titles}=  Get Card Badges Titles  card_locator=${card_locator}
+    Run Keyword And Continue On Failure  Lists Should Be Equal  ${card_badges_titles}  ${APPS_DICT}[${app_id}][badges]
+    Run Keyword If    $RH_BADGE_TITLE in $card_badges_titles
+    ...    Run Keyword And Continue On Failure  Page Should Contain Element    xpath:${card_locator}/${OFFICIAL_BADGE_XP}
+    [Return]  ${card_badges_titles}
+
+Open Get Started Sidebar And Return Status
+    [Arguments]  ${card_locator}
+    Click Element  xpath:${card_locator}
+    ${status}=  Run Keyword and Return Status  Wait Until Page Contains Element    xpath://div[contains(@class,'pf-c-drawer__panel-main')]
+    Sleep  1
```

Comment:
Why the sleep if you are using `Wait until page contains...`?

Reply:
@lugi0 sorry I closed this conversation by mistake without replying :) 
Basically the problem was that without wating at least 1 sec, the selenium driver wasn't able to catch all the links, even if the sidebar was open..I've tried alternative ways (e.g., Wait until specific elements..) but it was really unstable (sometimes working, sometimes not), so I decided to play the card of a Sleep.

Reply:
No worries, I thought that must've been the case. I would personally try smaller sleeps if possible (0.1s ~0.9s) to see if it is still stable while shaving off some time from execution, but lgtm regardless.

Reply:
I tried with 0.5s and it didn't work. Btw, I have a "sleeps" review as item for refactoring actually, so for sure I'll I review it again